.github/scripts/generate-build-matrices.py

```diff
@@ -23,7 +23,19 @@ def run_command(command: str) -> str:
     return result.stdout.strip()
 
 def get_module_list(ref: str) -> tuple[list[str], list[str]]:
-    changed_files = run_command(f"git diff --name-only {ref}").splitlines()
+    diff_output = run_command(f"git diff --name-status {ref}").splitlines()
+    
+    changed_files = []
+    for line in diff_output:
+        parts = line.split('\t')
+        status = parts[0]
+        if status.startswith('R'):  # Renamed/moved file
```

Comment:
```suggestion
        if status[:1] == 'R':  # Renamed/moved file
```

This is very Python specific, and might not be worth using when the main language of the repo is Kotlin.